src/main/java/com/hazelcast/aws/AwsDiscoveryStrategyFactory.java

```diff
@@ -50,4 +59,84 @@ public DiscoveryStrategy newDiscoveryStrategy(DiscoveryNode discoveryNode, ILogg
         }
         return definitions;
     }
+
+    /**
+     * Checks if Hazelcast is running on an AWS EC2 instance.
+     * <p>
+     * Note that this method returns {@code false} for any ECS environment, since currently there is no way to auto-configure
+     * Hazelcast network interfaces (required for ECS).
+     * <p>
+     * To check if Hazelcast is running on EC2, we first check that the machine uuid starts with "ec2" or "EC2". There is
+     * a small chance that a non-AWS machine has uuid starting with the mentioned prefix. That is why, to be sure, we make
+     * an API call to a local, non-routable address http://169.254.169.254/latest/dynamic/instance-identity/. Finally, we also
+     * check if an IAM Role is attached to the EC2 instance, because without any IAM Role the Hazelcast AWS discovery won't work.
+     *
+     * @return true if running on EC2 Instance which has an IAM Role attached
+     * @see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/identify_ec2_instances.html
+     */
+    @Override
+    public boolean isAutoDetectionApplicable() {
+        return isRunningOnEc2() && !isRunningOnEcs();
+    }
+
+    private static boolean isRunningOnEc2() {
+        return uuidWithEc2Prefix() && instanceIdentityExists() && iamRoleAttached();
+    }
+
+    private static boolean uuidWithEc2Prefix() {
+        String uuidPath = "/sys/hypervisor/uuid";
+        if (new File(uuidPath).exists()) {
+            String uuid = readFileContents(uuidPath);
+            return uuid.startsWith("ec2") || uuid.startsWith("EC2");
+        }
+        return false;
+    }
+
+    static String readFileContents(String fileName) {
+        InputStream is = null;
+        try {
+            File file = new File(fileName);
+            byte[] data = new byte[(int) file.length()];
+            is = new FileInputStream(file);
+            is.read(data);
+            return new String(data, StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            throw new RuntimeException("Could not get " + fileName, e);
+        } finally {
+            IOUtil.closeResource(is);
+        }
+    }
+
+    private static boolean instanceIdentityExists() {
+        return isEndpointAvailable("http://169.254.169.254/latest/dynamic/instance-identity/");
+    }
+
+    private static boolean iamRoleAttached() {
+        try {
+            return isEndpointAvailable("http://169.254.169.254/latest/meta-data/iam/security-credentials/");
+        } catch (Exception e) {
+            LOGGER.warning("Hazelcast running on EC2 instance, but no IAM Role attached. Cannot use Hazelcast AWS discovery.");
+            LOGGER.finest(e);
+            return false;
+        }
+    }
+
+    static boolean isEndpointAvailable(String url) {
+        int timeoutInSeconds = 1;
```

Comment:
Is there any extra value in naming a variable `timeoutInSeconds` in order to pass it to a method called `*TimeoutSeconds`? ;)

Reply:
Updated, thanks for the comment